### ISSUE TITLE
Connection macros fail to expand

### ISSUE THREAD

**Issue by colttt** [cavaliercoder/libzbxpgsql#17]
in version 0.2.0 and 0.2.1 all/most of the items are working, but i get the following error very often for all items:

```
 30996:20150922:093459.364 item "morbo:pg.table.idx_scan[{$PG_CONN},zabbix]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.db.tup_deleted[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.table.n_tup_del[{$PG_CONN},postgres]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.db.tup_inserted[{$PG_CONN},{$PG_DB},postgres]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.db.tup_inserted[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.db.tup_returned[{$PG_CONN},{$PG_DB},postgres]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.db.tup_returned[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.setting[{$PG_CONN},{$PG_DB},archive_command]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.setting[{$PG_CONN},{$PG_DB},archive_mode]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.setting[{$PG_CONN},{$PG_DB},archive_timeout]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.setting[{$PG_CONN},{$PG_DB},array_nulls]" became not supported: Unknown error.
 30996:20150922:093459.364 item "morbo:pg.setting[{$PG_CONN},{$PG_DB},authentication_timeout]" became not supported: Unknown error.
```

here a grep for a specific item..

```
 30994:20150922:081957.187 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
 30996:20150922:083408.464 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became supported
 30995:20150922:083458.585 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
 30995:20150922:084904.863 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became supported
 30994:20150922:084954.941 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
 30996:20150922:090406.310 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became supported
 30993:20150922:090456.469 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
 30995:20150922:091907.890 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became supported
 30994:20150922:091957.976 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
 30993:20150922:093409.239 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became supported
 30994:20150922:093454.362 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
 30994:20150922:094910.581 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became supported
 30996:20150922:094955.646 item "morbo:pg.db.temp_files[{$PG_CONN},{$PG_DB},zabbix]" became not supported: Unknown error.
```

i will try to debug them, to get more informations

**Comment by cavaliercoder** [comment]
Are these logs from the server or agent?

I noticed recently that the server takes time to apply macros after starting. That is, if your agent requests configuration too early, the macros (`{$PG_CONN}` and `{$PG_DB}`) are not expanded causing the items to become unsupported. After a while, when the agent refreshes its config, the keys are expanded.

This doesn't make sense but I was able to reproduce the issue. I'll take a look into it again and confirm.

Please also let me know your findings.

**Comment by colttt** [comment]
yes this logs are from zabbix-server.
my zabbix-server run more than a week, so i think it must be expanded for a while. I also think that zabbix-server caches this macros?

Bur you are right, the macros are not expanded.. after a while it works, and than after a while the same, macros are not expanded.

**Comment by cavaliercoder** [comment]
What version of Zabbix are you using? I'm testing on 2.2.10.

**Comment by colttt** [comment]
Hello,
i use zabbix_version 2.2.9

**Comment by cavaliercoder** [comment]
I've reproduced this issue with items not related to libzbxpgsql. For this reason, I believe this to be a Zabbix bug and have raised a ticket: https://support.zabbix.com/browse/ZBX-10190

**Comment by colttt** [comment]
wow thanks for you feedback, i hope it will be solved soon

**Comment by cavaliercoder** [comment]
Issues seems to be resolved in Zabbix 2.4+
